python/lsst/obs/subaru/ampOffset.py

```diff
@@ -0,0 +1,150 @@
+        exp = exposure.clone()
+        bitMask = exp.mask.getPlaneBitMask(self.background.config.ignoredPixelMask)
+        self.log.debug(f"Ignoring mask planes: {', '.join(self.background.config.ignoredPixelMask)}")
+
+        # fit and subtract background
+        maskedImage = exp.getMaskedImage()
+        bg = self.background.fitBackground(maskedImage)
+        bgImage = bg.getImageF(self.background.config.algorithm, self.background.config.undersampleStyle)
+        maskedImage -= bgImage
+
+        # detect sources and update cloned exposure mask planes in-place
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        table = afwTable.SourceTable.make(schema)
+        # detection sigma, used for smoothing and to grow detections, is
+        # normally measured from the PSF of the exposure. As the PSF has not
+        # yet been measured at this stage of processing, sigma is instead set
+        # to an approximate value here which should be sufficient.
+        _ = self.detection.run(table=table, exposure=exp, sigma=2)
+
+        # safety check: do any pixels remain for amp offset estimation?
+        if (exp.mask.array & bitMask).all():
+            raise pipeBase.TaskError("All pixels masked: cannot calculate any amp offset corrections.")
+
+        # set up amp offset inputs
+        im = exp.image
+        im.array[(exp.mask.array & bitMask) > 0] = np.nan
+        amps = exp.getDetector().getAmplifiers()
+        ampEdgeOuter = self.config.ampEdgeInset + self.config.ampEdgeWidth
+        sctrl = afwMath.StatisticsControl()
+
+        # loop over each amp edge boundary to extract amp-edge step values
+        ampSteps = []
+        for ii in range(1, len(amps)):
+            ampA = im[amps[ii - 1].getBBox()].array
+            ampB = im[amps[ii].getBBox()].array
+            stripA = ampA[:, -ampEdgeOuter:][:, :self.config.ampEdgeWidth]
+            stripB = ampB[:, :ampEdgeOuter][:, -self.config.ampEdgeWidth:]
+            # run with warning catcher to prevent all-NaN slice RuntimeWarning
+            with warnings.catch_warnings():
+                warnings.filterwarnings('ignore', r'All-NaN (slice|axis) encountered')
+                edgeA = np.nanmedian(stripA, axis=1)
+                edgeB = np.nanmedian(stripB, axis=1)
+            edgeDiff = edgeB - edgeA
+            # compute rolling averages
+            edgeDiffSum = np.convolve(np.nan_to_num(edgeDiff), np.ones(self.config.ampEdgeWindow), 'same')
+            edgeDiffNum = np.convolve(~np.isnan(edgeDiff), np.ones(self.config.ampEdgeWindow), 'same')
+            edgeDiffAvg = edgeDiffSum / np.clip(edgeDiffNum, 1, None)
+            edgeDiffAvg[np.isnan(edgeDiff)] = np.nan
+            # take clipped mean of rolling average data as amp-edge step value
+            ampStep = afwMath.makeStatistics(edgeDiffAvg, afwMath.MEANCLIP, sctrl).getValue()
+            minFracBad = (1 - (np.sum(np.isnan(edgeDiffAvg))/len(edgeDiffAvg))) < self.config.ampEdgeMinFrac
```

Comment:
Add a couple of comments on what's being checked here.  (a) that the fraction of pixels is large enough (?) and (b) that the step size is less than the configured max.

Reply:
The following comments have been added here:
```
# perform a couple of do-no-harm safety checks:
# a) the fraction of unmasked pixel rows is > ampEdgeMinFrac,
# b) the absolute offset ADU value is < ampEdgeMaxOffset
```